src/plugins/telemetry/server/telemetry_config/get_telemetry_opt_in.ts

```diff
@@ -42,6 +38,8 @@ export const getTelemetryOptIn: GetTelemetryOptIn = ({
   // if enabled is true, return it
   if (savedOptIn === true) return savedOptIn;
 
+  // TODO: Should we split the logic below into another OptIn getter?
```

Comment:
I thought I already did enough refactoring to also tackle this... IMO, it could be handled as part of https://github.com/elastic/kibana/issues/135107